File: Codewars/Javascript/7_kyu/Sum of integers in string.js
// Description:
// Your task in this kata is to implement a function that calculates the sum of the integers inside a string. For example, in the string "The30quick20brown10f0x1203jumps914ov3r1349the102l4zy dog", the sum of the integers is 3635.

// Note: only positive integers will be tested.

// My Solution:
function sumOfIntegersInString(s) {
    return s.split(/[^0-9]/g).map((e) => parseFloat(e)).filter(Boolean).reduce((a, b) => a + b, 0)
}

// The Boolean filter trick is very useful in this case

File: Codewars/Go/7_kyu/Sum of integers in string.go
// Your task in this kata is to implement a function that calculates the sum of the integers inside a string. For example, in the string "The30quick20brown10f0x1203jumps914ov3r1349the102l4zy dog", the sum of the integers is 3635.

// Note: only positive integers will be tested.

// My Solution:
package kata

import (
  "regexp"
  "strconv"
)

func SumOfIntegersInString(s string) int {
  re := regexp.MustCompile("[^0-9]+")
  parts := re.Split(s, -1)
  
  res := 0
  
  for _, part := range parts{
    if part != "" {
      num, err := strconv.Atoi(part)
      if err == nil {
        res += num
      }
    }
  }
  
  return res
}